pkg/processor/runtime/python/py/_nuclio_wrapper_as_server.py

```diff
@@ -0,0 +1,308 @@
+        server_sock.setsockopt(
+            # Level: Applies the option at the socket level
+            socket.SOL_SOCKET,
+            # Option: Allows reuse of the address/port
+            socket.SO_REUSEADDR,
+            # Value: Enables the option (0 would disable it)
+            1,
+        )
+        server_sock.bind((self._host, self._port))
+        server_sock.listen(self._max_connections)
+        server_sock.setblocking(False)
+
+        self._logger.info(f"Python wrapper server started on {self._host}:{self._port}, "
+                          f"max_connections: {self._max_connections}")
+
+        # Register server socket with default os selector (epoll for Linux)
+        self.selector.register(server_sock, selectors.EVENT_READ)
+        self.connections[server_sock.fileno()] = server_sock
+        try:
+            while True:
+                events = await self.select_events()
+                # Now awaiting the select operation
+                for key, event in events:
+                    sock = key.fileobj
+                    if sock == server_sock:
+                        # Accept new connection
+                        client_sock, addr = server_sock.accept()
+                        self._logger.info(f"Accepted connection from {addr}")
+                        client_sock.setblocking(False)
+                        self.connections[client_sock.fileno()] = client_sock
+                        task = self._loop.create_task(self._process_connection(client_sock))
+                        self.tasks[client_sock.fileno()] = task
+                    else:
+                        self._logger.error(f"Unexpected event: {event}")
+                await asyncio.sleep(0.2)
+        except KeyboardInterrupt:
+            self._logger.info("Shutting down server")
+        finally:
+            self._logger.info("Closing all connections")
+            for sock in self.connections.values():
+                sock.close()
+
+    async def select_events(self, timeout=0.1):
+        # Run the selector.select in the default executor (this is a blocking operation)
+        events = await self._loop.run_in_executor(None, self.selector.select, timeout)
+        return events
+
+    async def initialize(self):
+
+        # call init_context
+        await self._initialize_context()
+
+        # register to the SIGUSR1 and SIGUSR2 signals, used to signal termination/draining respectively
+        self._register_to_signal()
+
+        await self._send_data_on_control_socket({
+            'kind': 'wrapperInitialized',
+            'attributes': {'ready': 'true'}
+        })
+
+    async def _process_connection(self, sock):
+        """Process all events for a single connection."""
+        # signal start
+        self._logger.info("Signalling connection processing start")
+        await self._write_packet_to_processor(sock, 's')
+        self._logger.info(f"Event processing started for socket")
```

Comment:
Also maybe debug logs?
With a single socket `info` makes sense because it's only at the beginning.